### PR TITLE
YARN-11068. Update transitive log4j2 dependency to 2.17.1

### PR DESCRIPTION
**Pull request by jojochuang**
### Description of PR
Instead of excluding log4j2, perhaps we should try force its version to 2.17.1, which is not vulnerable to the log4shell issues.

### For code changes:

- [ ] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 59s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  shelldocs  |   0m  1s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  17m 38s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  27m  7s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  24m 25s |  |  trunk passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  compile  |  20m 46s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  mvnsite  |  26m 29s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   8m 19s |  |  trunk passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   8m 22s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  shadedclient  |  38m 52s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 28s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  24m 51s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  23m 40s |  |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javac  |  23m 40s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  20m 48s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |  20m 48s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |  21m 58s |  |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  xml  |   0m  3s |  |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   8m 20s |  |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   8m 12s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  shadedclient  |  39m 39s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 1038m  6s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3963/1/artifact/out/patch-unit-root.txt) |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 32s |  |  The patch does not generate ASF License warnings.  |
|  |   | 1329m 53s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.yarn.csi.client.TestCsiClient |
|   | hadoop.hdfs.rbfbalance.TestRouterDistCpProcedure |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3963/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3963 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell xml shellcheck shelldocs |
| uname | Linux f544242ab00f 4.15.0-162-generic #170-Ubuntu SMP Mon Oct 18 11:38:05 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / e18c73d360b622182cd9066db77a0f3f6d96c6ff |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3963/1/testReport/ |
| Max. process+thread count | 3079 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-applications-catalog/hadoop-yarn-applications-catalog-webapp . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3963/1/console |
| versions | git=2.25.1 maven=3.6.3 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by jojochuang** [comment]
The failed tests are not reproducible locally.

File: LICENSE-binary [hunk 1]
```diff
@@ -321,6 +321,7 @@ org.apache.kerby:kerby-pkix:1.0.1
 org.apache.kerby:kerby-util:1.0.1
 org.apache.kerby:kerby-xdr:1.0.1
 org.apache.kerby:token-provider:1.0.1
+org.apache.solr:solr-solrj:8.8.2
```
Comment (aajisaka) [inline LICENSE-binary:324]: I think this line can be removed because the solr jar file is not included the binary tarball.

Comment (jojochuang) [reply LICENSE-binary:324]: Hi Akira, it's actually bundled inside  the application catalog war file: hadoop-yarn-project/target/hadoop-yarn-project-3.4.0-SNAPSHOT/share/hadoop/yarn/hadoop-yarn-applications-catalog-webapp-3.4.0-SNAPSHOT.war

I am not familiar with this subproject... should we include dependencies from this war file?

Comment (aajisaka) [reply LICENSE-binary:324]: Thank you for the response.

```
% unzip -l ./hadoop-3.4.0-SNAPSHOT/share/hadoop/yarn/hadoop-yarn-applications-catalog-webapp-3.4.0-SNAPSHOT.war | grep solr
  2364784  11-19-2021 18:24   WEB-INF/lib/solr-solrj-8.8.2.jar
```

Given the jar file is actually in the binary tarball, we must add it in the LICENSE-binary file.

**Review by aajisaka — APPROVED** [review]

**Comment by jojochuang** [comment]
Thaks!

**Pull request merged**